### ISSUE TITLE
Deployment not starting any pod

### ISSUE THREAD

**Issue by torqu3e** [kelseyhightower/kubernetes-the-hard-way#462]
Everything up to DNS addon was followed exactly (except disk size for the compute instance being 50 GB instead of 200), applying the coredns.yaml gives the exact same output, the deployment and service get created but no pod starts. This is on GCP. 

```
$ kubectl get all -A
NAMESPACE     NAME                 TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)         AGE
default       service/kubernetes   ClusterIP   10.32.0.1    <none>        443/TCP         47h
kube-system   service/kube-dns     ClusterIP   10.32.0.10   <none>        53/UDP,53/TCP   23s

NAMESPACE     NAME                      DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
kube-system   deployment.apps/coredns   2         0         0            0           23s

```
```
$ kubectl get events -w
LAST SEEN   TYPE      REASON                    OBJECT          MESSAGE
29m         Normal    Starting                  node/worker-0   Starting kubelet.
29m         Normal    Starting                  node/worker-0   Starting kube-proxy.
29m         Warning   InvalidDiskCapacity       node/worker-0   invalid capacity 0 on image filesystem
29m         Normal    NodeAllocatableEnforced   node/worker-0   Updated Node Allocatable limit across pods
29m         Normal    NodeHasSufficientDisk     node/worker-0   Node worker-0 status is now: NodeHasSufficientDisk
29m         Normal    NodeHasSufficientMemory   node/worker-0   Node worker-0 status is now: NodeHasSufficientMemory
29m         Normal    NodeHasNoDiskPressure     node/worker-0   Node worker-0 status is now: NodeHasNoDiskPressure
29m         Normal    NodeHasSufficientPID      node/worker-0   Node worker-0 status is now: NodeHasSufficientPID
28m         Normal    Starting                  node/worker-0   Starting kubelet.
28m         Warning   InvalidDiskCapacity       node/worker-0   invalid capacity 0 on image filesystem
28m         Normal    NodeHasSufficientDisk     node/worker-0   Node worker-0 status is now: NodeHasSufficientDisk
28m         Normal    NodeHasSufficientMemory   node/worker-0   Node worker-0 status is now: NodeHasSufficientMemory
28m         Normal    NodeHasNoDiskPressure     node/worker-0   Node worker-0 status is now: NodeHasNoDiskPressure
28m         Normal    NodeHasSufficientPID      node/worker-0   Node worker-0 status is now: NodeHasSufficientPID
28m         Normal    NodeAllocatableEnforced   node/worker-0   Updated Node Allocatable limit across pods
28m         Normal    Starting                  node/worker-0   Starting kube-proxy.
29m         Normal    Starting                  node/worker-1   Starting kubelet.
29m         Normal    Starting                  node/worker-1   Starting kube-proxy.
29m         Warning   InvalidDiskCapacity       node/worker-1   invalid capacity 0 on image filesystem
29m         Normal    NodeHasSufficientDisk     node/worker-1   Node worker-1 status is now: NodeHasSufficientDisk
29m         Normal    NodeHasSufficientMemory   node/worker-1   Node worker-1 status is now: NodeHasSufficientMemory
29m         Normal    NodeHasNoDiskPressure     node/worker-1   Node worker-1 status is now: NodeHasNoDiskPressure
29m         Normal    NodeHasSufficientPID      node/worker-1   Node worker-1 status is now: NodeHasSufficientPID
29m         Normal    NodeAllocatableEnforced   node/worker-1   Updated Node Allocatable limit across pods
28m         Normal    Starting                  node/worker-1   Starting kubelet.
28m         Warning   InvalidDiskCapacity       node/worker-1   invalid capacity 0 on image filesystem
28m         Normal    NodeHasSufficientDisk     node/worker-1   Node worker-1 status is now: NodeHasSufficientDisk
28m         Normal    NodeHasSufficientMemory   node/worker-1   Node worker-1 status is now: NodeHasSufficientMemory
28m         Normal    NodeHasNoDiskPressure     node/worker-1   Node worker-1 status is now: NodeHasNoDiskPressure
28m         Normal    NodeHasSufficientPID      node/worker-1   Node worker-1 status is now: NodeHasSufficientPID
28m         Normal    NodeAllocatableEnforced   node/worker-1   Updated Node Allocatable limit across pods
28m         Normal    Starting                  node/worker-1   Starting kube-proxy.
29m         Normal    Starting                  node/worker-2   Starting kubelet.
29m         Warning   InvalidDiskCapacity       node/worker-2   invalid capacity 0 on image filesystem
29m         Normal    Starting                  node/worker-2   Starting kube-proxy.
29m         Normal    NodeHasSufficientDisk     node/worker-2   Node worker-2 status is now: NodeHasSufficientDisk
29m         Normal    NodeHasSufficientMemory   node/worker-2   Node worker-2 status is now: NodeHasSufficientMemory
29m         Normal    NodeHasNoDiskPressure     node/worker-2   Node worker-2 status is now: NodeHasNoDiskPressure
29m         Normal    NodeHasSufficientPID      node/worker-2   Node worker-2 status is now: NodeHasSufficientPID
29m         Normal    NodeAllocatableEnforced   node/worker-2   Updated Node Allocatable limit across pods
28m         Normal    Starting                  node/worker-2   Starting kubelet.
28m         Warning   InvalidDiskCapacity       node/worker-2   invalid capacity 0 on image filesystem
28m         Normal    NodeHasSufficientDisk     node/worker-2   Node worker-2 status is now: NodeHasSufficientDisk
28m         Normal    NodeHasSufficientMemory   node/worker-2   Node worker-2 status is now: NodeHasSufficientMemory
28m         Normal    NodeHasNoDiskPressure     node/worker-2   Node worker-2 status is now: NodeHasNoDiskPressure
28m         Normal    NodeHasSufficientPID      node/worker-2   Node worker-2 status is now: NodeHasSufficientPID
28m         Normal    NodeAllocatableEnforced   node/worker-2   Updated Node Allocatable limit across pods
28m         Normal    Starting                  node/worker-2   Starting kube-proxy.

```
```
$ kubectl get componentstatuses
NAME                 STATUS    MESSAGE             ERROR
controller-manager   Healthy   ok                  
scheduler            Healthy   ok                  
etcd-0               Healthy   {"health":"true"}   
etcd-1               Healthy   {"health":"true"}   
etcd-2               Healthy   {"health":"true"}   

```
```
$ kubectl get nodes
NAME       STATUS   ROLES    AGE    VERSION
worker-0   Ready    <none>   3h3m   v1.12.0
worker-1   Ready    <none>   3h3m   v1.12.0
worker-2   Ready    <none>   3h3m   v1.12.0
```
I decided to skip the coredns part and just run busybox or alpine or any other pod, again the deployment gets created but nothing happens from there on out.
```$ kubectl describe deployment.apps/coredns -n kube-system
Name:                   coredns
Namespace:              kube-system
CreationTimestamp:      Fri, 17 May 2019 15:20:18 -0400
Labels:                 k8s-app=kube-dns
                        kubernetes.io/name=CoreDNS
Annotations:            kubectl.kubernetes.io/last-applied-configuration:
                          {"apiVersion":"extensions/v1beta1","kind":"Deployment","metadata":{"annotations":{},"labels":{"k8s-app":"kube-dns","kubernetes.io/name":"C...
Selector:               k8s-app=kube-dns
Replicas:               2 desired | 0 updated | 0 total | 0 available | 0 unavailable
StrategyType:           RollingUpdate
MinReadySeconds:        0
RollingUpdateStrategy:  1 max unavailable, 1 max surge
Pod Template:
  Labels:           k8s-app=kube-dns
  Service Account:  coredns
  Containers:
   coredns:
    Image:       coredns/coredns:1.2.2
    Ports:       53/UDP, 53/TCP, 9153/TCP
    Host Ports:  0/UDP, 0/TCP, 0/TCP
    Args:
      -conf
      /etc/coredns/Corefile
    Limits:
      memory:  170Mi
    Requests:
      cpu:        100m
      memory:     70Mi
    Liveness:     http-get http://:8080/health delay=60s timeout=5s period=10s #success=1 #failure=5
    Environment:  <none>
    Mounts:
      /etc/coredns from config-volume (ro)
  Volumes:
   config-volume:
    Type:        ConfigMap (a volume populated by a ConfigMap)
    Name:        coredns
    Optional:    false
OldReplicaSets:  <none>
NewReplicaSet:   <none>
Events:          <none>
```
```
$ kubectl get deployment.apps/coredns -o yaml -n kube-system
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"extensions/v1beta1","kind":"Deployment","metadata":{"annotations":{},"labels":{"k8s-app":"kube-dns","kubernetes.io/name":"CoreDNS"},"name":"coredns","namespace":"kube-system"},"spec":{"replicas":2,"selector":{"matchLabels":{"k8s-app":"kube-dns"}},"strategy":{"rollingUpdate":{"maxUnavailable":1},"type":"RollingUpdate"},"template":{"metadata":{"labels":{"k8s-app":"kube-dns"}},"spec":{"containers":[{"args":["-conf","/etc/coredns/Corefile"],"image":"coredns/coredns:1.2.2","imagePullPolicy":"IfNotPresent","livenessProbe":{"failureThreshold":5,"httpGet":{"path":"/health","port":8080,"scheme":"HTTP"},"initialDelaySeconds":60,"successThreshold":1,"timeoutSeconds":5},"name":"coredns","ports":[{"containerPort":53,"name":"dns","protocol":"UDP"},{"containerPort":53,"name":"dns-tcp","protocol":"TCP"},{"containerPort":9153,"name":"metrics","protocol":"TCP"}],"resources":{"limits":{"memory":"170Mi"},"requests":{"cpu":"100m","memory":"70Mi"}},"securityContext":{"allowPrivilegeEscalation":false,"capabilities":{"add":["NET_BIND_SERVICE"],"drop":["all"]},"readOnlyRootFilesystem":true},"volumeMounts":[{"mountPath":"/etc/coredns","name":"config-volume","readOnly":true}]}],"dnsPolicy":"Default","serviceAccountName":"coredns","tolerations":[{"effect":"NoSchedule","key":"node-role.kubernetes.io/master"},{"key":"CriticalAddonsOnly","operator":"Exists"}],"volumes":[{"configMap":{"items":[{"key":"Corefile","path":"Corefile"}],"name":"coredns"},"name":"config-volume"}]}}}}
  creationTimestamp: "2019-05-17T19:20:18Z"
  generation: 1
  labels:
    k8s-app: kube-dns
    kubernetes.io/name: CoreDNS
  name: coredns
  namespace: kube-system
  resourceVersion: "138579"
  selfLink: /apis/apps/v1/namespaces/kube-system/deployments/coredns
  uid: ce464285-78d8-11e9-8152-42010af0000b
spec:
  progressDeadlineSeconds: 2147483647
  replicas: 2
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      k8s-app: kube-dns
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        k8s-app: kube-dns
    spec:
      containers:
      - args:
        - -conf
        - /etc/coredns/Corefile
        image: coredns/coredns:1.2.2
        imagePullPolicy: IfNotPresent
        livenessProbe:
          failureThreshold: 5
          httpGet:
            path: /health
            port: 8080
            scheme: HTTP
          initialDelaySeconds: 60
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 5
        name: coredns
        ports:
        - containerPort: 53
          name: dns
          protocol: UDP
        - containerPort: 53
          name: dns-tcp
          protocol: TCP
        - containerPort: 9153
          name: metrics
          protocol: TCP
        resources:
          limits:
            memory: 170Mi
          requests:
            cpu: 100m
            memory: 70Mi
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            add:
            - NET_BIND_SERVICE
            drop:
            - all
          procMount: Default
          readOnlyRootFilesystem: true
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /etc/coredns
          name: config-volume
          readOnly: true
      dnsPolicy: Default
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: coredns
      serviceAccountName: coredns
      terminationGracePeriodSeconds: 30
      tolerations:
      - effect: NoSchedule
        key: node-role.kubernetes.io/master
      - key: CriticalAddonsOnly
        operator: Exists
      volumes:
      - configMap:
          defaultMode: 420
          items:
          - key: Corefile
            path: Corefile
          name: coredns
        name: config-volume
status: {}
```

What am I doing wrong? Or what did I miss? I've scoured the web/stackoverflow etc and haven't found a solution to this.
The resolvconf file is set to the one used by ubuntu's resolver, so that is not an issue.

**Comment by dineshba** [comment]
Can you check the event specific to your deployment?
```bash
kubectl describe deploy coredns -n kube-system
```

>Note: Events in describe section will not show all the events. It will show only the latest events. If there are not events. May be deleting and creating the deployment will give the events. Events will definitely help us debug further.

**Comment by torqu3e** [comment]
Second from bottom code block in my initial report is the describe on coredns. There are no events shown. Interestingly, I started from scratch again and this time did not change the disk size from 200G and everything worked just fine.

I will at some point in the future try replicating my initial scenario and see if I missed something.

**Comment by ggaurav10** [comment]
The coredns deployment's `progressDeadlineSeconds` is set to max: 
```
spec:
  progressDeadlineSeconds: 2147483647
```

The actual issue in creating pods might be masked because of this.

**Comment by daftping** [comment]
I had a similar issue. In my case, it was due to broken kube-scheduler. If pods stuck in the pending state try to schedule them manually adding nodeName.
`kubectl create -f test.yaml`
```
apiVersion: v1
kind: Pod
metadata:
  name: test
spec:
  containers:
  - image: busybox
    name: test
    command:
    - sleep
    - "8400"
  nodeName: worker-0
```
If pod gets running, most probably it's scheduler issue.
ssh to controller-0 and check service logs
`journalctl --unit kube-scheduler.service`
In my case, there were authentication issues in logs. I found that my kube-scheduler.kubeconfig was missing the user part.
